probes/http/http.go

```diff
@@ -213,7 +219,13 @@ func isClientTimeout(err error) bool {
 }
 
 // httpRequest executes an HTTP request and updates the provided result struct.
-func (p *Probe) doHTTPRequest(req *http.Request, result *probeResult, resultMu *sync.Mutex) {
+func (p *Probe) doHTTPRequest(ireq *http.Request, result *probeResult, resultMu *sync.Mutex) {
+	req := ireq
+
+	if len(p.requestBody) >= largeBodyThreshold {
+		req = req.Clone(context.Background())
```

Comment:
It's not right to use background context here. Instead of that, use original request's context itself.


req = req.Clone(req.Context())

Reply:
Oops, cut and pasted from a different function, where that context wasn't set yet, without thinking. Fixed.